.github/workflows/umbral-pre.yml

```diff
@@ -106,7 +106,7 @@ jobs:
         rust:
           - stable
         python:
-          - "3.10"
+          - "3.12"
```

Comment:
AFAIK this gh action builds and runs tests (please correct me if I'm mistaken), so if we are claiming to support 3.8 all the way up to 3.12, should this python definition actually be a matrix with value `["3.8", "3.12"]` instead of a single python version.

Basically, I'm worried about us simply building & testing with 3.12 and then claiming we support older versions. eg. if we add a python 3.12 specific feature, that isn't compatible with older version of python, then the build/test can pass but we aren't actually supporting older versions.

I know that internally we are the only real users of rust-umbral, but it is published separately - so it's good to be consistent with what the library claims to support.